gather crates into workspace

euclio opened the pull request:
This enables running commands across all the crates at once, as well as making the crates share a Cargo.lock.

antoyo commented on .github/workflows/CI.yml:
```diff
@@ -56,20 +56,4 @@ jobs:
-      - name: "relm: build async example"
-        run: cargo build --manifest-path relm-examples/examples/async/Cargo.toml
-
-      - name: "relm: build webkit-test example"
```
The webkit example was kept separate so that if it fails because the crate is outdated or something, it's easy to see.

euclio replied:
Got it, I can move it back to its own crate. Would you like me to keep the other examples separate as well? The `buttons-attribute` test appeared to be duplicated.

euclio commented:
This is ready for another review. I pulled the webkit example back into its own crate. Also, I noticed the main crate's tests weren't passing so I fixed them and added them to CI.

euclio commented:
@antoyo Rebased. I decided to leave the examples as-is in their own crates. CI doesn't test them, just builds them. I think the main useful change here is that the main relm crate gets tested in CI now.

euclio commented:
We could also modify CI to just test and build everything in one go using the `--all` flag, but I wasn't sure if that's a change you want to make.
